### DISCUSSION TITLE
List all interfaces for a device

### DISCUSSION THREAD

**Discussion by digital-fortress** [netbox-community/netbox#16984] [Help Wanted!]
How can I get the list of all interfaces on  a device to do a bulk update on them.

/api/dcim/devices/5056/interfaces/
The requested page does not exist.

**Comment by AADIP** [comment]
`/api/dcim/interfaces/?device_id=5056` use the `interfaces` endpoint :)

**Comment by aascedu** [comment]
Hello @digital-fortress , have you found a way ? I think the api doesn't have access to the interfaces... Even in the data model, i can't find a way to get the interfaces ip and name to execute a script inside netbox...

Please help !

**Reply by digital-fortress** [reply]
I had found a way, but I forgot :(

**Reply by aascedu** [reply]
> I had found a way, but I forgot :(

I found a way to explore the models, which was in the docs all along : ./manage.py shell

Thanks anyways

**Comment by candlerb** [comment]
It should be something like `/api/dcim/interfaces/?device_id=xxx`

**Comment by aascedu** [comment]
Yes thank you, I hadn’t played enough with models in django as well as the
netbox models to understand EVERYTHING was linked !
